QRealTime_dialog.py

```diff
@@ -263,35 +293,36 @@ def updateLayer(self,layer,dataDict,geoField):
         type=layer.geometryType()
         geo=['POINT','LINE','POLYGON']
         layerGeo=geo[type]
-        
         uuidList = self.getUUIDList(self.processingLayer)
-
+        print("uuidlist is ",uuidList)
+        uuidList = list(dict.fromkeys(uuidList))
         newQgisFeatures = []
         fieldError = None
         for odkFeature in dataDict:
-            try:
-                qgisFeature = QgsFeature()
-                odkFeature=dict(odkFeature)
-                print('dict is',odkFeature)
-                wktGeom = self.guessWKTGeomType(odkFeature[geoField])
-                print (wktGeom)
-                if wktGeom[:3] != layerGeo[:3]:
-                    continue
-                qgisGeom = QgsGeometry.fromWkt(wktGeom)
-                print('geom is',qgisGeom)
-                qgisFeature.setGeometry(qgisGeom)
-                qgisFeature.initAttributes(len(QgisFieldsList))
-                for fieldName, fieldValue in six.iteritems(odkFeature):
-                    try:
-                        qgisFeature.setAttribute(QgisFieldsList.index(fieldName),fieldValue)
-                    except:
-                        fieldError = fieldName
+            if odkFeature['instanceID'] not in uuidList:
+                try:
+                    qgisFeature = QgsFeature()
+                    #odkFeature=dict(odkFeature)
+                    print('dict is',odkFeature)
+                    wktGeom = self.guessWKTGeomType(odkFeature[geoField])
+                    print ("WKTGeom is",wktGeom)
+                    if wktGeom[:3] != layerGeo[:3]:
+                        continue
+                    qgisGeom = QgsGeometry.fromWkt(wktGeom)
+                    print('geom is',qgisGeom)
+                    qgisFeature.setGeometry(qgisGeom)
+                    qgisFeature.initAttributes(len(QgisFieldsList))
+                    for fieldName, fieldValue in six.iteritems(odkFeature):
```

Comment:
i think there is no need to use six.iteritems

Reply:
yes sir, the same result can be obtained by using odkFeature.items() I'll make the necessary changes.